### PR TITLE
Connection management

### PR DESCRIPTION
**Pull request by Fraser999**
[Review on Reviewable](https://reviewable.io/reviews/dirvine/routing/133)

### CODE REVIEW

**Comment by dirvine** [comment]
Reviewed files:
- src/generic_sendable_type.rs @ r1
- src/lib.rs @ r1
- src/messages.rs @ r2
- src/messages/close_peer_lost.rs @ r2
- src/messages/connect_success.rs @ r2
- src/node_interface.rs @ r1

---

<sup>**[src/routing_node.rs, line 272 \[r2\]](https://reviewable.io:443/reviews/dirvine/routing/133#-Jo06b8QwyZYlBiGnNbB)** ([raw file](https://github.com/dirvine/routing/blob/b42e87b1fa96c64b989d73889ac1e4cce7e9a53f/src/routing_node.rs#L272)):</sup>
I think what we need here is `debug!` as opposed to `panic!` as panic will terminate the program in release. debug will allow testing if that is what we wish to achieve. No use of `unwrap` or calling `panic!` direct is a probably the safest route if in any doubt. If we avoid these we avoid crashes. 

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/133)

**Comment by dirvine** [comment]
Reviewed files:
- src/routing_table.rs @ r2
- src/test_utils/messages_util.rs @ r2

---

<sup>**[src/routing_node.rs, line 310 \[r2\]](https://reviewable.io:443/reviews/dirvine/routing/133#-Jo0GSK_xk_fAqW4aPuD)** ([raw file](https://github.com/dirvine/routing/blob/b42e87b1fa96c64b989d73889ac1e4cce7e9a53f/src/routing_node.rs#L310)):</sup>
lets have a mumble in the morning after the catch up, to go over the big picture here.

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/133)

File: src/routing_node.rs [hunk 1]
```diff
@@ -251,16 +254,59 @@ impl<F> RoutingNode<F> where F: Interface {
...
+        }
+        // TODO - it would be more efficient if routing_table.drop_node returned a bool to indicate
+        // whether the dropped node was in the close group or not - we wouldn't need to compare
+        // routing_table.our_close_group before and after to find out.
+        let close_group_before = BTreeSet::<NameType>::from_iter(
+            self.routing_table.our_close_group().iter().map(
+                |ref node_info| node_info.id.clone()));
+        self.routing_table.drop_node(&peer_id);
+        let close_group_after = self.routing_table.our_close_group();
+        self.all_connections.1.remove(&peer_id);
```
Comment (inetic) [inline src/routing_node.rs]: Has it not been removed already on line 271 ?

**Comment by inetic** [comment]
Apart from the line routing_node.rs:283 which might be redundant this looks correct to me.

**Comment by dirvine** [comment]
Closing but hold in branch.

**Pull request closed without merging**